src/go/k8s/controllers/redpanda/redpanda_controller_common_test.go

```diff
@@ -0,0 +1,41 @@
+package redpanda
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+const (
+	K8sInstanceLabelKey  = "app.kubernetes.io/instance"
+	K8sNameLabelKey      = "app.kubernetes.io/name"
+	K8sComponentLabelKey = "app.kubernetes.io/component"
+	K8sManagedByLabelKey = "app.kubernetes.io/managed-by"
+
+	EnvHelmReleaseNameKey = "REDPANDA_HELM_RELEASE_NAME"
+)
+
+var UpdateEventFilter = predicate.Funcs{
+	CreateFunc:  func(e event.CreateEvent) bool { return false },
+	UpdateFunc:  func(e event.UpdateEvent) bool { return true },
+	DeleteFunc:  func(e event.DeleteEvent) bool { return false },
+	GenericFunc: func(e event.GenericEvent) bool { return false },
+}
+
+var DeleteEventFilter = predicate.Funcs{
+	CreateFunc:  func(e event.CreateEvent) bool { return false },
+	UpdateFunc:  func(e event.UpdateEvent) bool { return false },
+	DeleteFunc:  func(e event.DeleteEvent) bool { return true },
+	GenericFunc: func(e event.GenericEvent) bool { return false },
+}
+
+// TODO check assumptions here
+// Check to see if the release name of a helm chart matches the name of a redpanda object
+// this is by design for the operator
+func isValidReleaseName(releaseName string, redpandaNameList []string) bool {
+	for i := range redpandaNameList {
+		if releaseName == redpandaNameList[i] {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
Why `src/go/k8s/controllers/redpanda/redpanda_controller_common_test.go` is the copy of `src/go/k8s/controllers/redpanda/redpanda_controller_common.go`?